src/mmw/js/src/data_catalog/views.js

```diff
@@ -546,32 +533,81 @@ var ResultDetailsCuahsiView = ResultDetailsBaseView.extend({
         });
     },
 
-    showValuesRegion: function() {
-        if (!this.valuesRegion) {
-            // Don't attempt to display values if this view
-            // has been unloaded.
-            return;
+    showChartOrTable: function() {
+        if (this.model.get('mode') === 'table') {
+            this.ui.chartRegion.addClass('hidden');
+            this.ui.tableRegion.removeClass('hidden');
+        } else {
+            this.ui.chartRegion.removeClass('hidden');
+            this.ui.tableRegion.addClass('hidden');
+
+            if (!this.chartRegion.hasView()) {
+                this.chartRegion.show(new CuahsiChartView({
+                    collection: this.model.get('variables'),
+                }));
+            }
         }
+    }
+});
 
-        var mode = this.model.get('mode'),
-            variables = this.model.get('variables'),
-            view = mode === 'table' ?
-                   new CuahsiTableView({ collection: variables }) :
-                   new CuahsiChartView({ collection: variables });
+var CuahsiStatusView = Marionette.ItemView.extend({
+    template: resultDetailsCuahsiStatusTmpl,
 
-        this.valuesRegion.show(view);
+    modelEvents: {
+        'change:fetching change:error': 'render',
+    },
+});
+
+var CuahsiSwitcherView = Marionette.ItemView.extend({
+    template: resultDetailsCuahsiSwitcherTmpl,
+
+    ui: {
+        chartButton: '#cuahsi-button-chart',
+        tableButton: '#cuahsi-button-table',
+    },
+
+    events: {
+        'click @ui.chartButton': 'setChartMode',
+        'click @ui.tableButton': 'setTableMode',
+    },
+
+    modelEvents: {
+        'change:fetching change:mode': 'render',
+    },
+
+    templateHelpers: function() {
+        var id = this.model.get('id'),
```

Comment:
This little guy is causing the build to fail; it's never used.

Reply:
Ah good catch, will remove.